tests/functional/aws-node-sdk/test/mdSearch/versionEnabledSearch.js

```diff
@@ -0,0 +1,72 @@
+const s3Client = require('./utils/s3SDK');
+const { runAndCheckSearch, removeAllVersions, runIfMongo } =
+    require('./utils/helpers');
+
+const userMetadata = { food: 'pizza' };
+const updatedMetadata = { food: 'salad' };
+const masterKey = 'master';
+
+runIfMongo('Search in version enabled bucket', () => {
+    const bucketName = `versionedbucket${Date.now()}`;
+    const VersioningConfiguration = {
+        MFADelete: 'Disabled',
+        Status: 'Enabled',
+    };
+    before(done => {
+        s3Client.createBucket({ Bucket: bucketName }, err => {
+            if (err) {
+                return done(err);
+            }
+            return s3Client.putBucketVersioning({ Bucket: bucketName,
+                VersioningConfiguration }, err => {
+                if (err) {
+                    return done(err);
+                }
+                return s3Client.putObject({ Bucket: bucketName,
+                    Key: masterKey, Metadata: userMetadata },
+                    err => {
+                        // give ingestion pipeline some time
+                        // setTimeout(() => done(err), 45000);
```

Comment:
just curious, why were the timeouts originally used? Is this the same as the timeout applied in the npm run command `ft_search`?

Reply:
The current implementation of MD search queries the mongo MD backend, and, as such, it does not require the wait time. (@rahulreddy confirmation/elaborate this statement?)

`setTimeout` was originally used here when MD search was using Clueso. As for how it relates to the npm command timeout: the npm timeout is set to a larger number to make sure this timeout wait does not expire the test.